parser_test.go

```diff
@@ -2570,6 +2571,18 @@ func (s *testParserSuite) TestDDL(c *C) {
 		{"alter table t reorganize partition remove partition;", false, ""},
 		{"alter table t reorganize partition no_write_to_binlog remove into (partition p0 VALUES LESS THAN (1991));", true, "ALTER TABLE `t` REORGANIZE PARTITION NO_WRITE_TO_BINLOG `remove` INTO (PARTITION `p0` VALUES LESS THAN (1991))"},
 
+		// alter placement rules
+		{"ALTER TABLE t ALTER PARTITION p ADD PLACEMENT LABEL='+zone=zone1' ROLE=LEADER COUNT=1", true, "ALTER TABLE `t` ALTER PARTITION `p` ADD PLACEMENT LABEL='+zone=zone1' ROLE=LEADER COUNT=1"},
+		{"ALTER TABLE t ALTER PARTITION p ADD PLACEMENT LABEL='-zone=zone1' ROLE=LEADER COUNT=1", true, "ALTER TABLE `t` ALTER PARTITION `p` ADD PLACEMENT LABEL='-zone=zone1' ROLE=LEADER COUNT=1"},
+		{"ALTER TABLE t ALTER PARTITION p ADD PLACEMENT LABEL='+zone=zone1,-zone=zone2' ROLE=LEADER COUNT=1", true, "ALTER TABLE `t` ALTER PARTITION `p` ADD PLACEMENT LABEL='+zone=zone1,-zone=zone2' ROLE=LEADER COUNT=1"},
+		{"ALTER TABLE t ALTER PARTITION p ADD PLACEMENT LABEL='+zone=zone1' ROLE=FOLLOWER COUNT=1", true, "ALTER TABLE `t` ALTER PARTITION `p` ADD PLACEMENT LABEL='+zone=zone1' ROLE=FOLLOWER COUNT=1"},
+		{"ALTER TABLE t ALTER PARTITION p ADD PLACEMENT ROLE=LEARNER LABEL='+zone=zone1' COUNT=1", true, "ALTER TABLE `t` ALTER PARTITION `p` ADD PLACEMENT LABEL='+zone=zone1' ROLE=LEARNER COUNT=1"},
+		{"ALTER TABLE t ALTER PARTITION p ADD PLACEMENT COUNT=1 LABEL='+zone=zone1' ROLE=VOTER", true, "ALTER TABLE `t` ALTER PARTITION `p` ADD PLACEMENT LABEL='+zone=zone1' ROLE=VOTER COUNT=1"},
+		{"ALTER TABLE t ALTER PARTITION p ADD PLACEMENT LABEL= ROLE=follower COUNT=1;", false, ""},
+		{"ALTER TABLE t ALTER PARTITION p ADD PLACEMENT LABEL='+zone=gh' ROLE=follower COUNT=-1;", false, ""},
+		{"ALTER TABLE t ALTER PARTITION p ADD PLACEMENT LABEL='+zone=gh' ROLE=follower COUNT=0;", false, ""},
+		{"ALTER TABLE t ALTER PARTITION p", false, ""},
+
```

Comment:
Add a test case that has duplicate LABEL, ROLE or COUNT.

Reply:
done